samples/Samples.HttpMessageHandler/TcpPortProvider.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Threading;
+
+namespace Datadog.Trace.TestHelpers
+{
+    /// <summary>
+    /// Helper class that tries to provide unique ports numbers across processes and threads in the same machine.
+    /// Used avoid port conflicts in concurrent tests that use the Agent, IIS, HttpListener, HttpClient, etc.
+    /// This class cannot guarantee a port is actually available, but should help avoid most conflicts.
+    /// </summary>
+    public static class TcpPortProvider
```

Comment:
Can we reuse the `TcpPortProvider` in `Datadog.Trace.TestHelpers` instead of duplicating the code?

Reply:
I would like to, except it messes up this test that *specifically* tests the GAC because the test project brings in a separate `System.Net.Http.dll`. Should I attempt to refactor that project or is this one-off code duplication okay?

Reply:
Or should I separate the GAC test from this sample project?

Reply:
I moved `TcpPortProvider` to `Datadog.Core.Tools` so that it can be shared more easily without extra baggage from the `Datadog.Trace.TestHelpers` library.